File: src/main/java/com/dv/telegram/command/HelpCommand.java
package com.dv.telegram.command;

import com.dv.telegram.WikiBot;

public class HelpCommand extends BasicBotCommand {

    @Override
    public String getName() {
        return HelpCommand.class.getSimpleName();
    }

    @Override
    public String getDescription(WikiBot bot) {
        return String.format(
            "`%s %s <commandName>` — получить помощь по команде с названием `<commandName>`.",
            bot.getBotName(),
            getCommandText()
        );
    }

    @Override
    public boolean useMarkdownInResponse() {
        return true;
    }

    @Override
    public String getDefaultCommandName() {
        return "/helpCommand";
    }

    @Override
    public String getResponse(String text, WikiBot bot) {
        int commandStartIndex = text.indexOf(getCommandText());
        if (commandStartIndex < 0) {
            return unknownCommandResponse();
        }

        int commandEndIndex = commandStartIndex + getCommandText().length();
        if (commandEndIndex >= text.length()) {
            return unknownCommandResponse();
        }

        String commandText = text.substring(commandEndIndex).trim();

        BotCommand botCommand = bot
            .getSpecialCommands()
            .getCommand(commandText);

        if (botCommand == null) {
            return unknownCommandResponse();
        }

        return String.format(
            "*%s*\n%s",
            botCommand.getCommandText(),
            botCommand.getDescription(bot)
        );
    }

    private String unknownCommandResponse() {
        return "Неизвестное имя команды.";
    }
}

File: src/main/kotlin/com/dv/telegram/command/HelpCommand.kt
package com.dv.telegram.command

import com.dv.telegram.WikiBot
import org.telegram.telegrambots.meta.api.objects.Update

class HelpCommand : BasicBotCommand() {
    override val name: String = javaClass.simpleName

    override fun getDescription(bot: WikiBot) =
        "`${bot.botName} $commandText <commandName>` — получить помощь по команде с названием `<commandName>`."

    override fun useMarkdownInResponse() = true

    override val defaultCommandName: String
        get() = "/helpCommand"

    override fun getResponse(text: String, bot: WikiBot, update: Update): String {
        val commandStartIndex = text.indexOf(commandText)
        if (commandStartIndex < 0) {
            return unknownCommandResponse()
        }

        val commandEndIndex = commandStartIndex + commandText.length
        if (commandEndIndex >= text.length) {
            return unknownCommandResponse()
        }

        val commandText = text.substring(commandEndIndex).trim()

        val botCommand = bot
            .specialCommands
            .getCommand(commandText)
            ?: return unknownCommandResponse()

        return "*${botCommand.commandText}*\n${botCommand.getDescription(bot)}"
    }

    private fun unknownCommandResponse() = "Неизвестное имя команды."
}
